### ISSUE TITLE
UnRar Issue - Failed to Convert CBR to CBZ

### ISSUE THREAD

**Issue by JasonMSims** [evilhero/mylar#1521]
I've combed through the open and closed issues as well as the forum, and although I've found similar issues, they all appear to be MacOS related.  I'm running Mylar on Ubuntu 16.04.  I updated UnRar to 5.40 and adjusted the settings in lib/comictaggerlib/ct_settings/settings.ini accordingly:

```
[settings]
check_for_new_version = False
rar_exe_path = /usr/local/bin/unrar
unrar_exe_path = /usr/local/bin/unrar
send_usage_stats = False
```

The error logs received are as follows:
```
2017-01-02 12:38:04	INFO	[POST-PROCESSING] Manual post-processing completed for 2 issues.
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Post-Processing ABORTED.
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Failed to move /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr - check directories and manually re-run.
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING][move] /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr ... to ... /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)/Detective Comics 937 (2016).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] odir src : /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Renaming /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Destination: /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)/Detective Comics 937 (2016).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Source: /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	INFO	[POST-PROCESSING][DIRECTORY-CHECK] Found comic directory: /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] New Filename: Detective Comics 937 (2016)
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Original Extension: .cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Original Filename: Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] ext:.cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] ofilename: Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] odir: /home/media/Downloads/comics
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] ofilename:Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Unable to write metadata successfully - check mylar.log file. Attempting to continue without tagging...
2017-01-02 12:38:04	WARNING	[META-TAGGER][COMIC-TAGGER][CBR-TO-CBZ] Failed to convert cbr to cbz - check permissions on folder : /opt/mylar/cache and/or the location where Mylar is trying to tag the files from.
2017-01-02 12:38:04	DEBUG	[META-TAGGER] Absolute path to script: /usr/bin/python
2017-01-02 12:38:04	DEBUG	[META-TAGGER] Executing command: ['/usr/bin/python', '/opt/mylar/comictagger.py', '-e', '--delete-rar', u'/opt/mylar/cache/mylar_ZD6sXI/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr']
2017-01-02 12:38:04	DEBUG	[META-TAGGER] Enabling ComicTagger script: ['/usr/bin/python', '/opt/mylar/comictagger.py'] with options: ['-e', '--delete-rar', u'/opt/mylar/cache/mylar_ZD6sXI/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr']
2017-01-02 12:38:04	DEBUG	[META-TAGGER] CBL Tagging enabled.
2017-01-02 12:38:04	DEBUG	[META-TAGGER] CR Tagging enabled.
2017-01-02 12:38:04	DEBUG	[META-TAGGER] ComicTagger 1.20.1 [ninjas.walk.alone / SHURIKEN] being used - using personal ComicVine API key supplied via mylar.
2017-01-02 12:38:04	DEBUG	[META-TAGGER] Running the ComicTagger Add-on for Mylar
2017-01-02 12:38:04	DEBUG	[META-TAGGER] comicpath : /opt/mylar/cache/mylar_ZD6sXI
2017-01-02 12:38:04	DEBUG	[META-TAGGER] sabnzbdscriptpath : /opt/mylar
2017-01-02 12:38:04	DEBUG	[META-TAGGER] downloadpath : /home/media/Downloads/comics
2017-01-02 12:38:04	DEBUG	[META-TAGGER] scriptname : Mylar.py
2017-01-02 12:38:04	DEBUG	[META-TAGGER] Paths / Locations:
2017-01-02 12:38:04	DEBUG	[META-TAGGER] Filename is : /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	ComicTagger Path location for internal comictagger.py set to : /opt/mylar/comictagger.py
2017-01-02 12:38:04	DEBUG	[META-TAGGER] dirName:/home/media/Downloads/comics
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Metatagging enabled - proceeding...
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] New format: $Series $Issue ($Year)
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Not an annual - removing from filename parameters
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Comic Version: v3
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Comic Location: /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Year: 2016
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Series: Detective Comics
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Publisher: DC Comics
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Issue Year : 2016
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Pretty Comic Issue is : 937
2017-01-02 12:38:04	DEBUG	Zero level supplement set to 0x. Issue will be set as : 937
2017-01-02 12:38:04	DEBUG	issue detected greater than 100
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Zero Suppression set to : 0x
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] issuenumOG: 937
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] issueid: 541161
2017-01-02 12:38:04	INFO	[POST-PROCESSING] [2/2] Starting Post-Processing for Detective Comics issue: 937
2017-01-02 12:38:04	INFO	[DUPECHECK] Duplication detection returned no hits. This is not a duplicate of anything that I have scanned in as of yet.
2017-01-02 12:38:04	INFO	[DUPECHECK] Duplicate check for /home/media/Downloads/comics/Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Post-Processing ABORTED.
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Failed to move /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr - check directories and manually re-run.
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING][move] /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr ... to ... /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)/Detective Comics 938 (2016).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] odir src : /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Renaming /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Destination: /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)/Detective Comics 938 (2016).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Source: /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	INFO	[POST-PROCESSING][DIRECTORY-CHECK] Found comic directory: /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] New Filename: Detective Comics 938 (2016)
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Original Extension: .cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Original Filename: Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] ext:.cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] ofilename: Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] odir: /home/media/Downloads/comics
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] ofilename:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:04	DEBUG	[POST-PROCESSING] Unable to write metadata successfully - check mylar.log file. Attempting to continue without tagging...
2017-01-02 12:38:04	WARNING	[META-TAGGER][COMIC-TAGGER][CBR-TO-CBZ] Failed to convert cbr to cbz - check permissions on folder : /opt/mylar/cache and/or the location where Mylar is trying to tag the files from.
2017-01-02 12:38:03	DEBUG	[META-TAGGER] Absolute path to script: /usr/bin/python
2017-01-02 12:38:03	DEBUG	[META-TAGGER] Executing command: ['/usr/bin/python', '/opt/mylar/comictagger.py', '-e', '--delete-rar', u'/opt/mylar/cache/mylar_J_j9mV/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr']
2017-01-02 12:38:03	DEBUG	[META-TAGGER] Enabling ComicTagger script: ['/usr/bin/python', '/opt/mylar/comictagger.py'] with options: ['-e', '--delete-rar', u'/opt/mylar/cache/mylar_J_j9mV/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr']
2017-01-02 12:38:03	DEBUG	[META-TAGGER] CBL Tagging enabled.
2017-01-02 12:38:03	DEBUG	[META-TAGGER] CR Tagging enabled.
2017-01-02 12:38:03	DEBUG	[META-TAGGER] ComicTagger 1.20.1 [ninjas.walk.alone / SHURIKEN] being used - using personal ComicVine API key supplied via mylar.
2017-01-02 12:38:03	DEBUG	[META-TAGGER] Running the ComicTagger Add-on for Mylar
2017-01-02 12:38:03	DEBUG	[META-TAGGER] comicpath : /opt/mylar/cache/mylar_J_j9mV
2017-01-02 12:38:03	DEBUG	[META-TAGGER] sabnzbdscriptpath : /opt/mylar
2017-01-02 12:38:03	DEBUG	[META-TAGGER] downloadpath : /home/media/Downloads/comics
2017-01-02 12:38:03	DEBUG	[META-TAGGER] scriptname : Mylar.py
2017-01-02 12:38:03	DEBUG	[META-TAGGER] Paths / Locations:
2017-01-02 12:38:03	DEBUG	[META-TAGGER] Filename is : /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	ComicTagger Path location for internal comictagger.py set to : /opt/mylar/comictagger.py
2017-01-02 12:38:03	DEBUG	[META-TAGGER] dirName:/home/media/Downloads/comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Metatagging enabled - proceeding...
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] New format: $Series $Issue ($Year)
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Not an annual - removing from filename parameters
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Comic Version: v3
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Comic Location: /media/MEDIAHDD/content/Comics/DC Comics/Detective Comics (2016)
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Year: 2016
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Series: Detective Comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Publisher: DC Comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Issue Year : 2016
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Pretty Comic Issue is : 938
2017-01-02 12:38:03	DEBUG	Zero level supplement set to 0x. Issue will be set as : 938
2017-01-02 12:38:03	DEBUG	issue detected greater than 100
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Zero Suppression set to : 0x
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] issuenumOG: 938
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] issueid: 543724
2017-01-02 12:38:03	INFO	[POST-PROCESSING] [1/2] Starting Post-Processing for Detective Comics issue: 938
2017-01-02 12:38:03	INFO	[DUPECHECK] Duplication detection returned no hits. This is not a duplicate of anything that I have scanned in as of yet.
2017-01-02 12:38:03	INFO	[DUPECHECK] Duplicate check for /home/media/Downloads/comics/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] There are 2 files found that match on your watchlist, 0 do not match anything.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Now Checking if the issue also resides in one of the storyarc's that I am watching.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][SUCCESSFUL MATCH: Detective Comics-91098] Match verified for Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][ISSUE-VERIFY][Issue Year MATCH] Issue Year of 2016 is a match to the year found in the filename of : 2016
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][ISSUE-VERIFY][Lone Volume FAILURE] Volume label of v3 indicates that there is more than one volume for this series, but the one on your watchlist has no volume label set.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][ISSUE-VERIFY] Now checking against Detective Comics-91098
2017-01-02 12:38:03	DEBUG	[MATCH: Detective Comics] Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	Possible Alternate Names to match against (if necessary): ['127372873872871091383 abdkhjhskjhkjdhakajhf']
2017-01-02 12:38:03	INFO	Series_Name: Detective Comics --- WatchComic: Detective Comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] No corresponding issue # found for 18058
2017-01-02 12:38:03	DEBUG	[MATCH: Detective Comics] Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	Possible Alternate Names to match against (if necessary): ['127372873872871091383 abdkhjhskjhkjdhakajhf']
2017-01-02 12:38:03	INFO	Series_Name: Detective Comics --- WatchComic: Detective Comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] There are 1 files found that match on your watchlist, 1 do not match anything.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Now Checking if the issue also resides in one of the storyarc's that I am watching.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][SUCCESSFUL MATCH: Detective Comics-91098] Match verified for Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][ISSUE-VERIFY][Issue Year MATCH] Issue Year of 2016 is a match to the year found in the filename of : 2016
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][ISSUE-VERIFY][Lone Volume FAILURE] Volume label of v3 indicates that there is more than one volume for this series, but the one on your watchlist has no volume label set.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING][ISSUE-VERIFY] Now checking against Detective Comics-91098
2017-01-02 12:38:03	DEBUG	[MATCH: Detective Comics] Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	Possible Alternate Names to match against (if necessary): ['127372873872871091383 abdkhjhskjhkjdhakajhf']
2017-01-02 12:38:03	INFO	Series_Name: Detective Comics --- WatchComic: Detective Comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] No corresponding issue # found for 18058
2017-01-02 12:38:03	DEBUG	[MATCH: Detective Comics] Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	Possible Alternate Names to match against (if necessary): ['127372873872871091383 abdkhjhskjhkjdhakajhf']
2017-01-02 12:38:03	INFO	Series_Name: Detective Comics --- WatchComic: Detective Comics
2017-01-02 12:38:03	INFO	I have located 2 files that I should be able to post-process. Continuing...
2017-01-02 12:38:03	DEBUG	series title possibly: Detective Comics
2017-01-02 12:38:03	DEBUG	issue verified as : 937
2017-01-02 12:38:03	DEBUG	issue_position: 2
2017-01-02 12:38:03	DEBUG	possible_issuenumbers: [{'position': 2, 'validcountchk': False, 'number': '937', 'mod_position': 17}]
2017-01-02 12:38:03	DEBUG	highest_series_position: 3
2017-01-02 12:38:03	DEBUG	year verified as: 2016
2017-01-02 12:38:03	DEBUG	year verified as : 2016
2017-01-02 12:38:03	DEBUG	sctd: 2017
2017-01-02 12:38:03	DEBUG	datecheck: [{'date': '2016', 'position': 3, 'mod_position': 21}]
2017-01-02 12:38:03	DEBUG	date: 2016
2017-01-02 12:38:03	DEBUG	checking date : 2016
2017-01-02 12:38:03	DEBUG	Scanner group tag discovered: Minutemen-Drunk Monk
2017-01-02 12:38:03	INFO	[FILENAME]: Detective Comics 937 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	DEBUG	series title possibly: Detective Comics
2017-01-02 12:38:03	DEBUG	issue verified as : 938
2017-01-02 12:38:03	DEBUG	issue_position: 2
2017-01-02 12:38:03	DEBUG	possible_issuenumbers: [{'position': 2, 'validcountchk': False, 'number': '938', 'mod_position': 17}]
2017-01-02 12:38:03	DEBUG	highest_series_position: 3
2017-01-02 12:38:03	DEBUG	year verified as: 2016
2017-01-02 12:38:03	DEBUG	year verified as : 2016
2017-01-02 12:38:03	DEBUG	sctd: 2017
2017-01-02 12:38:03	DEBUG	datecheck: [{'date': '2016', 'position': 3, 'mod_position': 21}]
2017-01-02 12:38:03	DEBUG	date: 2016
2017-01-02 12:38:03	DEBUG	checking date : 2016
2017-01-02 12:38:03	DEBUG	Scanner group tag discovered: Minutemen-Drunk Monk
2017-01-02 12:38:03	INFO	[FILENAME]: Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr
2017-01-02 12:38:03	INFO	there are 2 files.
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Manual Run initiated
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] Not using SABnzbd : Manual Run
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] nzb folder: /home/media/Downloads/comics
2017-01-02 12:38:03	DEBUG	[POST-PROCESSING] nzb name: Manual Run
2017-01-02 12:38:03	INFO	Starting postprocessing for : Manual Run
```

Permissions for the cache directory are:
```
drwxrwxrwx 16 media media    20480 Jan  2 12:38 cache
```

Permissions generated for the temp folders are:
```
drwx------ 2 media media   4096 Jan  2 12:38 mylar_ZD6sXI
drwx------ 2 media media   4096 Jan  2 12:38 mylar_J_j9mV

```

Up until recently, everything worked as expected, but stopped within the last couple of weeks.

**Comment by evilhero** [comment]
If I had to guess I would suspect permissions - if the user running mylar isn't the user media,  and it's in a group it might have something to do with the problem. If you're using a 3rd party app to run mylar that can cause permissions problems as well.

There hasn't been anything going into dev that would impact meta-tagging in the last few weeks - unless you're running the master branch. 

Does it actually complete the cbr to cbz conversion and the cbz file is in the cache directory?

**Comment by JasonMSims** [comment]
The user running mylar is media, no 3rd party apps are being introduced into the workflow and I am on the latest development commit.  It does not complete the conversion - it simply moves the cbr to the cache dir and aborts when it first tries to unrar.  No cbz file is generated.

**Comment by evilhero** [comment]
I believe 5.40 was the branch of unrar that was having issues with extracting due to invalid timestamps within the compressed file. I think if you're running the latest 5.40 (at the time of all the errors with the macs and such, it was at 5.40b6 I believe or something simliar - mine's a joke I run 4.00 but it never fails lol)

What you can try to help narrow it down, is in the mylar/cmtagmylar.py file - there are 2 commented out logger lines that might provide more help as to the error:
On lines 189&190 (approximately):
```
           #logger.info(out)
           #logger.info(err)
```
just uncomment so it's:
```
            logger.info(out)
            logger.info(err)
```
Delete the cmtagmylar.pyc file (just in case) and try to metatag again - it should spit out more info (if you run it in console mode instead of as a daemon, you'll get the errors breaking on the screen instead of being flung into limbo and not trapped by the logger)

**Comment by JasonMSims** [comment]
Below is the extra info logged:

```
2017-01-03 14:23:58	WARNING	[META-TAGGER][COMIC-TAGGER][CBR-TO-CBZ] Failed to convert cbr to cbz - check permissions on folder : /opt/mylar/cache/ and/or the location where Mylar is trying to tag the files from.
2017-01-03 14:23:58	INFO	None
2017-01-03 14:23:58	INFO	readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 1]
readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 2]
readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 3]
readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 4]
readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 5]
readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 6]
readArchiveFile(): [file is not expected size: 2055416 vs 0] /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbr:Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk)/Detective Comics 938-000.jpg [attempt # 7]
Error while copying to /opt/mylar/cache/mylar_90N4lu/Detective Comics 938 (2016) (2 covers) (digital) (Minutemen-Drunk Monk).cbz: 
Archive failed to export!
```

So it appears that at some point it's assuming the filesize to be zero and then error'ing out when comparing the actual filesize to that assumption.  Manually unrar'ing the file via the cli works fine, but still no dice when trying to get mylar/comictagger to unrar.

**Comment by evilhero** [comment]
Is the location where the cache is located full ? The error size 0 would indicate that it's unable to compare it to the proper size as it can't actually unrar the file due to space limitations (ie. the hdd is full).

That error is actually CT failing, I'm unsure if it's related to your unrar version or as I mentioned above the possibility of storage size being full.

**Comment by JasonMSims** [comment]
I have ~85GB free currently on that HDD.

**Comment by evilhero** [comment]
Ok, well was just a best guess - then it has to do with the unrar that it's pulling against for some reason (or unrar2 since that's what it's trying to do it with from the code). This is happening to every file you try to meta-tag correct ? And what is the file-size / permissions on the cbr/rar file located within the cache directory ? 

Sorry for all the questions, I've honestly never seen that error occur until right now.

**Comment by JasonMSims** [comment]
Permissions: -rwxrw-r--
Filesize: 33.8MB

This is happening without exception to every cbr I attempt to post-process.  No worries at all about the volume of questions, I just appreciate you being responsive!

**Comment by evilhero** [comment]
If it's happening to every cbr you post-process, I would still think it's the version of unrar it's not liking. CT uses pyUNRaR, which is basically a python wrapper to your unrar binary on your OS. You said earlier you're running 5.40, which from the looks of it is the trial-version of unrar which has limitations and a different api which doesn't work with the pyUNRaR wrapper. Can you try upgrading that to a newer version ? 

According to the site, 5.45 is the latest version by source and you can grab it from [here](http://www.rarlab.com/rar/unrarsrc-5.4.5.tar.gz)

**Comment by JasonMSims** [comment]
I just compiled that source and the unrar head still shows it to be "UNRAR 5.40 freeware".  Continued with install, tried to post-process and same result as before.  Should I try to downgrade to an older unrar?

**Comment by evilhero** [comment]
Is it possible you're still pointing to the 5.40 unrar instead of the 5.45 ? I believe you have to copy the compiled rar/unrar into your /usr/bin (or wherever you have your binaries) directory. 

When you're in the compiled directory with the unrar/rar binary that you just compiled, doing the typical './unrar' should indicate the correct header ?

**Comment by evilhero** [comment]
You can try downgrading, yes. Is there any way you can view the information for one of those cbr files ? I'm wondering if they're in the rar format 3.x instead of a 2.x and that might be causing some problems - which would end up being the version of unrar being used again if it was lol

**Comment by JasonMSims** [comment]
I ran it as '.unrar' in the compiled directory.  Just removed everything, redownloaded, re-extracted, and recompiled to be sure.  The newly compiled header is still showing 5.40, even though it's supposedly the 5.4.5 source.

Good news though!  I found 4.2.4 [here](http://old-releases.ubuntu.com/ubuntu/pool/multiverse/u/unrar-nonfree/unrar_4.2.4-0.3_amd64.deb).  After removing 5.40 and installing 4.2.4, it's working!

**Comment by evilhero** [comment]
Yep, that's 2.9 - although weird that it has a 2.0, but that might be something totally unrelated.

Glad you got it working finally! I kinda figured it was due to the unrar version, but there's so many damned versions out there with very little information in regards to fixes it makes it really difficult to figure out which version is right / wrong for each user.

I'll bookmark that link just in case as I'm sure this will come up again at some point ;) Especially since I just upgraded my unrar to 5.40, hah!
